### DISCUSSION TITLE
AXAML `using:` vs `clr-namespace:`?

### DISCUSSION THREAD

**Discussion by MathiasLui** [AvaloniaUI/Avalonia#11835] [Q&A]
When I created an Avalonia MVVM Desktop Project, I got the default "MainWindow" view, which had an XML namespace like this:
`xmlns:vm="using:MyGreatApp.ViewModels"`
and then the DataType with something like this:
`x:DataType="vm:MainWindowViewModel"`

Now I wanted to create a UserControl, and I also created a ViewModel for it at the same place as the MainWindowViewModel, and I set it via `xmlns:vm="using:MyGreatApp.ViewModels"`, and then `x:DataType="vm:MyCustomViewModel"` - but this time it told me it couldn't find the type in the namespace and it didn't show the designer or any Intellisense, but it still compiled successfully. 

After a long while I replaced `using` with `clr-namespace` and it worked. Aren't they supposed to work the same, given that the ViewModels were in the same assembly?

**Comment by timunie** [comment]
Yeah, should work the same. Sometimes a fresh build is needed for the previewer to pickup

**Reply by MathiasLui** [reply]
I restarted VS and my PC several times and did several of the classic "Cleans" followed by "Rebuilds".
But as soon as I changed it to `clr-namespace`, it instantly worked.

**Reply by timunie** [reply]
which Avalonia version? If 11.0 RC1 or later, please provide a very minimal sample to reproduce.

**Reply by MathiasLui** [reply]
Yes, the newest (11), I'll try to make an example this evening (in ~4 hours or so), I don't think I did anything more than what I described

**Reply by timunie** [reply]
provide what you have so far.

**Reply by MathiasLui** [reply]
I just tried recreating the issue on another PC and it worked normally as expected. 

The code I meant is at home, though. As mentioned I'll look at it in a few hours.

**Reply by MathiasLui** [reply]
Update:

I feel stupid, I created a ViewModel for that UserControl, and one for the list inside of it. I changed it from `internal` to `public`, but of the wrong ViewModel, so just making it public fixed it.

Thanks for your answering though :D

**Reply by timunie** [reply]
Oh that's interesting. So clr accepts internal viewmodels? 👀

Will need to test it tomorrow

**Reply by MathiasLui** [reply]
Seems that way 👀

**Reply by timunie** [reply]
works for me on Rider. Probably a VS thing.
